src/parser/time.rs

```diff
@@ -66,35 +66,23 @@ pub(crate) fn parse_time_spec(cursor: &mut Cursor) -> TemporalResult<TimeSpec> {
     })
 }
 
+/// Parse an hour value.
+#[inline]
 pub(crate) fn parse_hour(cursor: &mut Cursor) -> TemporalResult<u8> {
-    let start = cursor.pos();
-    for _ in 0..2 {
-        let digit = cursor.abrupt_next()?;
-        assert_syntax!(digit.is_ascii_digit(), "Hour must be a digit.");
-    }
-    let hour_value = cursor
-        .slice(start, cursor.pos())
-        .parse::<u8>()
-        .map_err(|e| TemporalError::syntax().with_message(e.to_string()))?;
+    let first = cursor.next_digit()?;
+    let hour_value = first * 10 + cursor.next_digit()?;
     if !(0..=23).contains(&hour_value) {
         return Err(TemporalError::syntax().with_message("Hour must be in a range of 0-23"));
     }
     Ok(hour_value)
 }
 
 // NOTE: `TimeSecond` is a 60 inclusive `MinuteSecond`.
-/// Parse `MinuteSecond`
+/// Parse `MinuteSecond` value.
```

Comment:
Can you link to the corresponding grammar rule?

Reply:
Can do. Better documentation all the way around is definitely needed. Should we update the grammar rules as a whole in a follow-up PR or do you mind if I add them in on this PR?

Reply:
If you're planning on doing a PR only for documentation, you can move this change to that PR.